corehq/apps/app_manager/models.py

```diff
@@ -5599,21 +5602,50 @@ def domain_link(self):
         from corehq.apps.linked_domain.dbaccessors import get_domain_master_link
         return get_domain_master_link(self.domain)
 
-    def get_master_version(self):
+    @memoized
+    def get_master_app_briefs(self):
         if self.domain_link:
-            return get_master_app_version(self.domain_link, self.master)
+            return get_master_app_briefs(self.domain_link)
+        return []
 
     @property
     def master_is_remote(self):
         if self.domain_link:
             return self.domain_link.is_remote
 
-    def get_latest_master_release(self):
+    def get_latest_master_release(self, master_app_id):
         if self.domain_link:
-            return get_latest_master_app_release(self.domain_link, self.master)
+            return get_latest_master_app_release(self.domain_link, master_app_id)
         else:
             raise ActionNotPermitted
```

Comment:
I realize that you didn't write this, but the if/else doesn't match the rest of the code in this class. Could be

```
if self.domain_link:
   # do thing
raise ActionNotPermitted
```

Reply:
46e31c48c1c